react/src/js/components/Consonant/Helpers/__tests__/CardFilterer.spec.js

```diff
@@ -79,12 +79,12 @@ describe('utils/CardFilterer', () => {
 
         test('Event Sort', () => {
             const cards = [
-                { id: 1, contentArea: { dateDetailText: { startTime: '2021-01-01', endTime: '2021-01-02' } }, tags: ['event1'] },
-                { id: 2, contentArea: { dateDetailText: { startTime: '2021-01-03', endTime: '2029-01-04' } }, tags: ['event2'] },
+                { id: 1, contentArea: { dateDetailText: { startTime: '2021-01-01', endTime: '2021-01-02' } }, tags: ['event1'], footer: [{ left: []}] },
```

Comment:
We should add test coverage for our fallback logic for `startDate/endDate` when `card.footer[0].left[1]` may or may not exist, so we don’t throw errors or pick the wrong date. 

The key scenarios:

1. No `footer` at all – Confirm we gracefully use `contentArea.dateDetailText`.
2. `footer` exists but only [0] in left – Ensure no error and fallback works.
3. `footer[0].left[1]` fully present – Confirm it overrides contentArea.
4. Partial `footer[0].left[1]` – If one date is missing, ensure fallback grabs it from contentArea.